ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/IndirectVertxTimerUtil.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.p2p.discovery.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import io.vertx.core.Vertx;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/*
+ * In May of 2020 we had issues with some combination of vertx, circleci, and
+ * the machines it was running on no respecting timer delays.  The solution was
+ * to use JDK executors for those timer delays.
+ *
+ * This should be revisited at a future date to see if we still need this.
+ */
+public class IndirectVertxTimerUtil implements TimerUtil {
+
+  private final ScheduledExecutorService secheduledExecutor =
+      Executors.newSingleThreadScheduledExecutor();
+  private final AtomicLong nextId = new AtomicLong(0);
```

Comment:
Is this supposed to get incremented somewhere?

Reply:
Yep, missed it.